apps/wallet/src/ui/app/components/sui-apps/SuiApp.tsx

```diff
@@ -19,80 +18,142 @@ export type DAppEntry = {
-				{displayType === 'full' && <div className={st.description}>{description}</div>}
+		</div>
+	);
+}
 
-				{displayType === 'card' && <div className={st.link}>{originLabel}</div>}
+type ListViewProp = {
+	name: string;
+	icon?: string;
+	description: string;
+	tags?: string[];
+};
 
-				{displayType === 'full' && tags?.length && (
-					<div className={st.tags}>
+function ListView({ name, icon, description, tags }: ListViewProp) {
+	return (
+		<div className="bg-white group flex py-3.5 px-1.25 gap-3 item-center box-border rounded hover:bg-sui/10">
+			<ImageIcon src={icon || null} label={name} fallback={name} size="xl" circle />
+			<div className="flex flex-col gap-1 justify-center">
+				<Text variant="body" weight="semibold" color="sui-dark">
+					{name}
+				</Text>
+				<Text variant="subtitle" weight="normal" color="steel-darker">
+					{description}
+				</Text>
+				{tags?.length && (
+					<div className="flex flex-wrap gap-1">
 						{tags?.map((tag) => (
-							<div className={st.tag} key={tag}>
-								{tag}
+							<div
+								className="flex item-center justify-center px-1.5 py-0.5 border border-solid border-steel rounded"
+								key={tag}
+							>
+								<Text variant="captionSmall" weight="medium" color="steel-dark">
+									{tag}
+								</Text>
 							</div>
 						))}
 					</div>
 				)}
 			</div>
 		</div>
 	);
-	return (
+}
+
+export interface SuiAppProps {
+	name: string;
+	description: string;
+	link: string;
+	icon: string;
+	tags: string[];
+	permissionID?: string;
+	displayType: DisplayType;
+	openAppSite?: boolean;
+}
+
+export function SuiApp({
+	name,
+	description,
+	link,
+	icon,
+	tags,
+	permissionID,
+	displayType,
+	openAppSite,
+}: SuiAppProps) {
+	const [showDisconnectApp, setShowDisconnectApp] = useState(false);
+	const appUrl = getDAppUrl(link);
+
+	if (permissionID && showDisconnectApp) {
+		return (
+			<DisconnectApp
+				name={name}
+				link={link}
+				icon={icon}
+				permissionID={permissionID}
+				setShowDisconnectApp={setShowDisconnectApp}
+			/>
+		);
+	}
+
+	const AppDetails = (
 		<>
-			{permissionID && showDisconnectApp ? (
-				<DisconnectApp
-					name={name}
-					link={link}
-					icon={icon}
-					permissionID={permissionID}
-					setShowDisconnectApp={setShowDisconnectApp}
-				/>
-			) : null}
-			{permissionID ? (
-				<div className={st.ecosystemApp} onClick={() => setShowDisconnectApp(true)}>
-					{AppDetails}
-				</div>
+			{displayType === 'full' ? (
+				<ListView name={name} description={description} icon={icon} tags={tags} />
 			) : (
-				<ExternalLink
-					href={appUrl?.toString() ?? link}
-					title={name}
-					className={st.ecosystemApp}
-					onClick={() => {
-						trackEvent('AppOpen', {
-							props: { name, source: 'AppPage' },
-						});
-					}}
-				>
-					{AppDetails}
-				</ExternalLink>
+				<CardView name={name} link={link} icon={icon} />
 			)}
 		</>
 	);
+
+	if (permissionID && !openAppSite) {
+		return (
+			<button
```

Comment:
We could use a div here, right?

Reply:
I use button for accessibility, I will change it

Reply:
Yeah we could add `role="button"`